src/main/java/hudson/plugins/distfork/DistForkCommand.java

```diff
@@ -154,8 +154,14 @@ public void run() {
 
                         if (!returnFiles.isEmpty() || returnZip!=null) {
                             stderr.println("Copying back files");
-                            for (Entry<String, String> e : returnFiles.entrySet())
-                                workDir.child(e.getValue()).copyToWithPermission(new FilePath(channel,e.getKey()));
+                            for (Entry<String, String> e : returnFiles.entrySet()) {
+                                FilePath tmp = new FilePath(channel, e.getKey() + ".tmp");
+                                FilePath actual = new FilePath(channel, e.getKey());
+                                workDir.child(e.getValue()).copyToWithPermission(tmp);
+                                if (actual.exists())
+                                    actual.delete();
```

Comment:
There is a risk of leaving the temporary file of this operation fails due to whatever reason. Likely it's not important